src/Command/MergeAssetsAndVariationsFilesCommand.php

```diff
@@ -256,4 +260,33 @@ private function hasValidFilePaths(string $assetsFilePath, string $variationsFil
 
         return $hasValidFilePath;
     }
+
+    /**
+     * Parse the whole variations CSV file and index them by Asset Code:
+     * This is to avoid to re-parse the whole variations file for each asset later.
+     */
+    private function indexVariationLinesByAssetCode(): void
+    {
+        $headers = $this->variationsReader->getHeaders();
+
+        foreach ($this->variationsReader as $variationLineNumber => $variationRow) {
+            $variationLine = array_combine($headers, $variationRow);
+
+            if ($variationLineNumber === 1) {
+                continue;
+            }
+
+            if (!$this->isHeaderValid($this->variationsReader, $variationRow)) {
+                continue;
+            }
+
+            $index = $this->getIndexationKey($variationLine['asset']);
+            $this->indexedVariations[$index][] = $variationLine;
+        }
+    }
+
+    private function getIndexationKey($assetCode): string
```

Comment:
is it stil needed ?

Reply:
It's to avoid magic/mysterious cast to in when asking for array key. Maybe a bit paranoid but this has no cost, I think we could keep it :3